digitaltwin_client/inc/digitaltwin_interface_client.h

```diff
@@ -287,7 +287,7 @@ MOCKABLE_FUNCTION(, DIGITALTWIN_CLIENT_RESULT, DigitalTwin_InterfaceClient_SetCo
 
   @returns  A DIGITALTWIN_CLIENT_RESULT.
 */
-MOCKABLE_FUNCTION(, DIGITALTWIN_CLIENT_RESULT, DigitalTwin_InterfaceClient_SendTelemetryAsync, DIGITALTWIN_INTERFACE_CLIENT_HANDLE, dtInterfaceClientHandle, const char*, telemetryName, const char*, messageData, DIGITALTWIN_CLIENT_TELEMETRY_CONFIRMATION_CALLBACK, telemetryConfirmationCallback, void*, userContextCallback);
+MOCKABLE_FUNCTION(, DIGITALTWIN_CLIENT_RESULT, DigitalTwin_InterfaceClient_SendTelemetryAsync, DIGITALTWIN_INTERFACE_CLIENT_HANDLE, dtInterfaceClientHandle, const char*, telemetryName, const unsigned char*, messageData, DIGITALTWIN_CLIENT_TELEMETRY_CONFIRMATION_CALLBACK, telemetryConfirmationCallback, void*, userContextCallback);
```

Comment:
>messageData [](start = 207, length = 11)

We need a messageDataLen on this one because we can't count on strlen() in a future binary binding.  And in CreateJsonForTelemetryMessage() call to STRING_construct_sprintf() we'll need the %*. or whatever the sprintf syntax is for setting number of chars.

Reply:
And similar feedback for other places in stack, like dt_interface_private.h, that interact with this.

---
In reply to: [333751205](https://github.com/Azure/azure-iot-sdk-c/pull/1245#discussion_r333751205) [](ancestors = 333751205)